### DISCUSSION TITLE
Don't enable TLS 1.0 and TLS 1.1 by default

### DISCUSSION THREAD

**Discussion by ghost** [curl/curl#8527] [General]
According to [RFC8996](https://datatracker.ietf.org/doc/html/rfc8996) TLS 1.0 and TLS 1.1 are deprecated.

Chrome, Microsoft Edge, Brave, Firefox and Safari have already disabled TLS 1.0 and TLS 1.1

Edit: Instead of completely disabling it, it is better to not enable by default, but allowing the user to override it

**Comment by dfandrich** [comment]
curl is used in many places where browsers don't go. I'm sure there are
internal company servers that are stuck on TLS 1.1, and will be for a while,
which applications still need to access. Setting --tlsv1.2 (or the equivalent)
by default is probably a good idea, but it should still be possible for people
to override it when necessary, in the same way that --insecure is sometimes
necessary but is not the default.

**Comment by ghost** [comment]
OK, sorry for the lack of thorough consideration.
By the way that RFC is "Best Current Practice" instead of "Standards Track"

**Comment by bagder** [comment]
In my opinion, it's too early for us to make that move. Anyone that really want to, can still build their own curl or TLS library to disable those options.
